x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/job_creator.ts

```diff
@@ -622,6 +622,66 @@ export class JobCreator {
     return JSON.stringify(this._datafeed_config, null, 2);
   }
 
+  private _initPerPartitionCategorization() {
+    if (this._job_config.analysis_config.per_partition_categorization === undefined) {
+      this._job_config.analysis_config.per_partition_categorization = {};
+    }
+    if (this._job_config.analysis_config.per_partition_categorization?.enabled === undefined) {
+      this._job_config.analysis_config.per_partition_categorization!.enabled = false;
+    }
+    if (this._job_config.analysis_config.per_partition_categorization?.stop_on_warn === undefined) {
+      this._job_config.analysis_config.per_partition_categorization!.stop_on_warn = false;
+    }
+  }
+
+  public get perPartitionCategorization() {
+    return this._job_config.analysis_config.per_partition_categorization?.enabled === true;
+  }
+
+  public set perPartitionCategorization(enabled: boolean) {
+    this._initPerPartitionCategorization();
+    this._job_config.analysis_config.per_partition_categorization!.enabled = enabled;
+  }
+
+  public get perPartitionStopOnWarn() {
+    return this._job_config.analysis_config.per_partition_categorization?.stop_on_warn === true;
+  }
+
+  public set perPartitionStopOnWarn(enabled: boolean) {
+    this._initPerPartitionCategorization();
+    this._job_config.analysis_config.per_partition_categorization!.stop_on_warn = enabled;
+  }
+
+  public get categorizationPerPartitionField(): string | null {
+    // looping through to find current partition_field name to prevent stale/syncing issue
+    // possible because partition_field_name has to have same value in every detector that uses the keyword mlcategory
+    const firstCategorizationDetector = this._detectors.find(
+      (d) => d.by_field_name === 'mlcategory'
+    );
+    if (
+      firstCategorizationDetector &&
+      'partition_field_name' in firstCategorizationDetector &&
+      firstCategorizationDetector.partition_field_name !== undefined
+    ) {
+      return firstCategorizationDetector.partition_field_name;
+    }
+    return null;
+  }
+
+  public set categorizationPerPartitionField(fieldName: string | null) {
+    if (fieldName === null) {
+      this._detectors.forEach((detector) => {
+        delete detector.partition_field_name;
+      });
+    } else {
+      if (this.categorizationPerPartitionField !== fieldName) {
```

Comment:
thinking about this a bit more, this check won't catch the situation where a second `mlcategory` detector has been added with a different partition field.
I think we're going to struggle to add nice UI behaviour in this situation without looping over the detectors whenever one is added to check for `mlcategory`.

Reply:
Just tested this and validation fails in a bad way when we have two categorisation detectors with different partition fields.
I think we should allow this to be configured in the UI, and we should fix validation so it displays the error in the correct way.

Reply:
After chatting to @droberts195 we agree that the advanced job wizard shouldn't have a global way to configure the partition field for all categorisation detectors.
Instead the user should do this themselves per detector.
This means the `categorizationPerPartitionField` can be moved out of the base `JobCreator` class and instead live in the derived `CategorizationJob` class.
It also means we can put the `partitionField` class variable back in as there will only ever be one detector in that class.